runtime/lua/vim/treesitter/language.lua

```diff
@@ -96,29 +102,27 @@ function M.add(lang, opts)
   lang = lang:lower()
 
   if vim._ts_has_language(lang) then
-    return
+    return true
   end
 
   if path == nil then
+    -- allow only safe language names when looking for libraries to load
     if not (lang and lang:match('[%w_]+') == lang) then
-      error("'" .. lang .. "' is not a valid language name")
+      return
     end
 
     local fname = 'parser/' .. lang .. '.*'
     local paths = api.nvim_get_runtime_file(fname, false)
     if #paths == 0 then
-      error("no parser for '" .. lang .. "' language, see :help treesitter-parsers")
+      return
     end
     path = paths[1]
   end
 
   if vim.endswith(path, '.wasm') then
-    if not vim._ts_add_language_from_wasm then
-      error(string.format("Unable to load wasm parser '%s': not built with ENABLE_WASMTIME ", path))
-    end
-    vim._ts_add_language_from_wasm(path, lang)
+    return vim._ts_add_language_from_wasm and vim._ts_add_language_from_wasm(path, lang)
```

Comment:
@lewis6991 This may prioritize a wasm parser over a native one even though we may not build with WASM and then fail.

On the other hand, we would happily have tried to load a `python.borkbork` even before that...